### DISCUSSION TITLE
MongoDB Geospatial Query Operators

### DISCUSSION THREAD

**Discussion by yogevlahyani** [prisma/prisma#22392] [Q&A]
### Question

For days I've been trying to understand how can I use Prisma to create a query that finds the following criteria:
Get 5 results of alerts within a specific location (lat, long) radius and a where clause of nested related document.

As far as I know, Prisma doesn't support MongoDB Geospatial Query Operators.

The where clause I'm trying to combine with the raw query:
```javascript
return this.prisma.alert.findMany({
      where: {
        status: 0,
        sensor: { infrastructure: { organizationId } },
      },
      take: 5,
    });
```

The raw query without the organizationId:
```javascript
this.prisma.alert.findRaw({
      filter: {
        status: 0,
        location: {
          $geoWithin: {
            $center: [[latitude, longitude], 1000],
          },
        },
      },
      options: {
        limit: 5,
      },
    });
```

My goal is to get the AlertModel as prisma returns it (without the _id ObjectId object comes from mongo, but a plain id as a string) that are only within the nested organization and within a radius of 1km of the provided `latitude` and `longitude`.

Is there a robust way of achieving that goal without making workarounds, hacks or working with different ORM/ODM?

### Information about Prisma Schema, Client Queries and Environment (optional)

- OS: macOS
- Database: MongoDB
- Node.js version: v18.17.1

Prisma schema:

```prisma
model Alert {
  id           String     @id @default(auto()) @map("_id") @db.ObjectId
  status       Int        @default(0)
  sensorId     String?    @db.ObjectId
  sensor       Sensor?    @relation(fields: [sensorId], references: [id])
  latitude     Float
  longitude    Float
  location     Json

  @@unique([sensorId])
  @@index([location], map: "2dsphere")
}

model Sensor {
  id                   String            @id @default(auto()) @map("_id") @db.ObjectId
  infrastructure       Infrastructure    @relation(fields: [infrastructureId], references: [id])
  infrastructureId     String            @db.ObjectId
  alerts               Alert[]
}

model Infrastructure {
  id               String              @id @default(auto()) @map("_id") @db.ObjectId
  organizationId   String              @db.ObjectId
  organization     Organization        @relation(fields: [organizationId], references: [id])
  sensors          Sensor[]
}

model Organization {
  id              String           @id @default(auto()) @map("_id") @db.ObjectId
  name            String
  infrastructures Infrastructure[]
}
```

**Comment by ludralph** [comment]
Hi @yogevlahyani 👋 

Thank you for raising this question.

As you may be aware, Prisma doesn't have full support for GeoSpatial queries with MongoDB. There is an open issue for this.

- https://github.com/prisma/prisma/issues/1798

In the issue, a user shared a [workaround](https://github.com/prisma/prisma/issues/1798#issuecomment-1697487437) on how they got a GeoJson sample working. Can you please check if that suits your use case? You can add a 👍 to the issue and also include use case so that our engineers can take it into consideration when implementing the feature.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise, please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by yogevlahyani** [reply]
I've seen the workaround and it's working but I can't filter with related models when using `findRaw`:

```javascript
filter: {
    sensor: { infrastructure: { organizationId } }
}
```

**Reply by ludralph** [reply]
I guess this is a limitation. It would be helpful If you could document this on the open issue so our engineers can have a look.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
